### PR TITLE
fix for the issue #390 (empty gpkg verification)

### LINKED ISSUE #390: AOI creation fails with empty gpkg
A valid gpkg containing no features raises error during creation of an AOI instance:

image

To reproduce:
```
python GDL.py mode=verify dataset.raw_data_csv="your/dir/data_with_empty_gpkg.csv"
dataset.bands=[1]
debug=False
```

Files (raster uses pre-existing raster in GDL's test data as dummy):
input data csv: [data_with_empty_gpkg.csv](https://github.com/NRCan/geo-deep-learning/files/10034339/data_with_empty_gpkg.csv)
ground truth gpkg: [buil_AB11-WV02-20100926-1.zip](https://github.com/NRCan/geo-deep-learning/files/10034343/buil_AB11-WV02-20100926-1.zip)

### PR DESCRIPTION
**Pull request by ms888ekb**
fixes #390 

- added a handler for the empty geopackage verification 
- added a corresponding unit test

### CODE REVIEW

**Review by CharlesAuthier — COMMENTED** [review]
Dont add '[tests/dataset/buil_AB11-WV02-20100926-1.zip](https://github.com/NRCan/geo-deep-learning/pull/407/files#diff-7ec9a513b62ed128a37d00acdb56ae90870c3acf956aa9b7dcce6d4b4cfe0c61)' to the PR except if @mpelchat04  have given you the OK.

**Comment by ms888ekb** [comment]
> Dont add '[tests/dataset/buil_AB11-WV02-20100926-1.zip](https://github.com/NRCan/geo-deep-learning/pull/407/files#diff-7ec9a513b62ed128a37d00acdb56ae90870c3acf956aa9b7dcce6d4b4cfe0c61)' to the PR except if @mpelchat04 have given you the OK.

@mpelchat04 would you give me OK please?

**Comment by remtav** [comment]
> > Dont add '[tests/dataset/buil_AB11-WV02-20100926-1.zip](https://github.com/NRCan/geo-deep-learning/pull/407/files#diff-7ec9a513b62ed128a37d00acdb56ae90870c3acf956aa9b7dcce6d4b4cfe0c61)' to the PR except if @mpelchat04 have given you the OK.
> 
> @mpelchat04 would you give me OK please?

If I'm not mistaken, our legal constraints about data only convern imagery. Moreover, this gpkg is empty. I don't see any problem with adding empty gpkgs to our test data. The only information we're making public is the name of the gpkg. I'd be surprised if there's an issue with that. I'll let @mpelchat04 decide :)

**Review by remtav — COMMENTED** [review]
Could you also address behavior during tiling for this issue? 
With the current implementation, the tiling process raises the following error:
```
  File "C:\Users\rtavon\PycharmProjects\geo-deep-learning\tiling_segmentation.py", line 639, in main
    tiler_pair = tiler.tiling_per_aoi(aoi, out_img_dir=tiling_dir_img, out_label_dir=tiling_dir_gt)
  File "C:\Users\rtavon\PycharmProjects\geo-deep-learning\tiling_segmentation.py", line 311, in tiling_per_aoi
    vec_dataset = GDLVectorDataset(aoi.label)
  File "C:\Users\rtavon\PycharmProjects\geo-deep-learning\dataset\create_dataset.py", line 212, in __init__
    self.index.insert(0, coords, 'vec')
  File "C:\Users\rtavon\Miniconda3\envs\geo_deep_env\lib\site-packages\rtree\index.py", line 462, in insert
    p_mins, p_maxs = self.get_coordinate_pointers(coordinates)
  File "C:\Users\rtavon\Miniconda3\envs\geo_deep_env\lib\site-packages\rtree\index.py", line 369, in get_coordinate_pointers
    raise RTreeError(
rtree.exceptions.RTreeError: Coordinates must not have minimums more than maximums
```

If a csv contains an AOI with an empty gpkg, I'd suggest we raise an error during tiling process to notify the user that "at least one ground truth file is empty. Run "verify" mode to identify these AOIs and remove them from input csv prior to tiling". This is a suggestion of course :)

Also, don't forget to add "fixes #[issue number]" in description of PR. Once the PR is merged, this tells github to close the related issue.

File: tiling_segmentation.py
Comment (remtav) [inline tiling_segmentation.py:634]: ok!

**Review by remtav — APPROVED** [review]

**Review by CharlesAuthier — APPROVED** [review]

**Pull request merged**
